api/envoy/api/v2/discovery.proto

```diff
@@ -191,6 +190,12 @@ message DeltaDiscoveryResponse {
   // in the DeltaDiscoveryRequest.
   repeated Resource resources = 2 [(gogoproto.nullable) = false];
 
+  // field id 3 IS available!
+
+  // Type URL for resources. Identifies the xDS API when muxing over ADS.
+  // Must be consistent with the type_url in the Any within 'resources' if 'resources' is non-empty.
+  string type_url = 4;
```

Comment:
I think this is fine (and will merge), because this is consistent with our previous approach. Do you think (forward looking) there is a case for mixed types in a single `DeltaDiscoveryResponse`? I.e. the ability to try and deliver an atomic-ish update?

Reply:
Hmm. Possibly. I think you would need it structured as "here is a series of updates, apply them in this order", although at that point you're just kind of saving on some overhead from multiple gRPC messages. To do a more seriously atomic thing, there would need to be some serious changes in the Envoy xDS consumers to support that.

Reply:
Although I guess the "batching" approach would give you the benefit of allowing the Envoy to return an all-or-nothing (N)ACK for the batch, which might simplify things for the server.